aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -585,3 +588,15 @@ def handle_electronic_convergence_not_achieved(self, calculation):
         action = f'reduced beta mixing from {mixing_beta} to {mixing_beta_new} and restarting from the last calculation'
         self.report_error_handled(calculation, action)
         return ProcessHandlerReport(True)
+
+    @process_handler(priority=610, exit_codes=[
```

Comment:
note that the priority has nothing to do with the exit code. By making it `610` this handler will now actually be ran before all others. I don't have a waterproof argument for where it should be otherwise, but I would put it close, probably just before the handler of the actual `ERROR_ELECTRONIC_CONVERGENCE_NOT_REACHED`, which has priority 410, so maybe we give this one 420.

Reply:
I actually had a question about this: does the `PwCalculation` ever have multiple exit codes? Just by glancing at the `PwParser`, doesn't the parsing stop once e.g.: 

https://github.com/aiidateam/aiida-quantumespresso/blob/f4528aff3165dbb8c598e8ef5ab91e81d9587b60/aiida_quantumespresso/parsers/pw.py#L105-L107

so even _if_ other exit codes could be added that are dealt with in the `PwBaseWorkChain` at a higher priority, will this happen? If not, does the priority really matter if the `PwCalculation` only has one exit code? Maybe I'm misunderstanding how this all works. I was looking into it when I had the ambition of reviewing this PR, but I went too deep down the rabbit hole and forgot what I was doing at some point. 😅

Reply:
> I actually had a question about this: does the `PwCalculation` ever have multiple exit codes? 

Any `Process` can only ever have a single exit code, that is a limitation imposed by `aiida-core`.

> so even _if_ other exit codes could be added that are dealt with in the `PwBaseWorkChain` at a higher priority, will this happen? If not, does the priority really matter if the `PwCalculation` only has one exit code?

It does matter, because process handlers return instances of `ProcessHandlerReport` which have an attribute called `do_break`. If that is set to `False` (which is the default) the `BaseRestartWorkChain` will continue calling other process handlers. So depending on the return values of process handlers the order in which they are executed *can* matter.

Reply:
Alright, thanks for the explanation! Will be looking closer at this in the coming month when reviewing the tutorial material on the topic.